core/src/main/java/org/infinispan/util/concurrent/CompletableFutures.java

```diff
@@ -36,6 +40,23 @@
       sink.setFuture(compoundSource);
    }
 
+   public static <T> CompletableFuture<T> connect(NotifyingFuture<T> source) {
```

Comment:
More than `connect`, I'd name this `asCompletableFuture` or `toCompletableFuture`. This is often the terminology used in Scala implicit conversions where there are tons of methods like this, with the added benefit that the conversions are implicit rather than you having to spell out the calls.